src/rust-macro-section/numeric-retrievals.md

```diff
@@ -0,0 +1,85 @@
+# Getting or combining subslices of bytes
+
+Some of the macros defined in the Win32 SDK are used to take smaller slices of bytes from larger ones. For example `LOWORD` and `HIWORD` to get the most and least significant `WORD`s from 32-bit integer types serving as packed data. Helpers like these are luckily rather trivial to implement and only require some bitmasking.
+
+## Getting window `LPARAM` values
+
+```rs
+/// Get the low order word as a signed 32-bit integer.
+#[inline(always)]
+pub fn get_x_lparam(lp: windows::Win32::Foundation::LPARAM) -> i32 {
+    lp.0 as i32
+}
```

Comment:
This looks wrong. *windowsx.h* has the `GET_X_LPARAM` defined as:

```c
#define GET_X_LPARAM(lp)                        ((int)(short)LOWORD(lp))
```

A corresponding Rust implementation should then be:

```rust
pub fn get_x_lparam(lp: windows::Win32::Foundation::LPARAM) -> i32 {
    lp.0 as i16 as i32
}
```

The `as i16` part truncates the value to the low word with the subsequent `as i32` sign-extending the value. This follows the rules for [type cast expressions](https://doc.rust-lang.org/reference/expressions/operator-expr.html#type-cast-expressions).

Reply:
Great catch, missef that one. I'll look into the corresponding behavior for C++ as well to make sure we don't change values if it sign-extends a negative value, then fix it when I'm home